### ISSUE TITLE
Unexpected output shape [1, 20, 21504] for YOLO11-OBB — should be [1, 21, ...]?

### ISSUE THREAD

**Issue by 0xF9M** [ultralytics/ultralytics#21603]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi,
I’m using the yolo11n-obb.pt model with 15 classes (DOTA-v1.0) and noticed that the output shape of the exported TorchScript model is [1, 20, 21504].
But I was expecting 21 channels in the first dimension — corresponding to:

    4 box coordinates (x, y, w, h)

    1 confidence score

    1 rotation angle

    15 class logits

That should give 21 total output channels, not 20.

Here is a minimal reproduction:

```
from ultralytics import YOLO
import torch

model = YOLO("yolo11n-obb.pt")
model.names

# {0: 'plane',
#  1: 'ship',
#  2: 'storage tank',
#  3: 'baseball diamond',
#  4: 'tennis court',
#  5: 'basketball court',
#  6: 'ground track field',
#  7: 'harbor',
#  8: 'bridge',
#  9: 'large vehicle',
#  10: 'small vehicle',
#  11: 'helicopter',
#  12: 'roundabout',
#  13: 'soccer ball field',
#  14: 'swimming pool'}

model.export(format="torchscript")
# Ultralytics 8.3.174 🚀 Python-3.13.5 torch-2.7.1+cu126 CPU (11th Gen Intel Core(TM) i7-11800H 2.30GHz)
# YOLO11n-obb summary (fused): 109 layers, 2,656,648 parameters, 0 gradients, 6.6 GFLOPs
#
# PyTorch: starting from 'yolo11n-obb.pt' with input shape (1, 3, 1024, 1024) BCHW and output shape(s) (1, 20, 21504) (5.5 MB)
#
# TorchScript: starting export with torch 2.7.1+cu126...
# TorchScript: export success ✅ 1.5s, saved as 'yolo11n-obb.torchscript' (10.9 MB)
#
# Export complete (1.9s)
# Results saved to /home/florian/Projects/ai_projects/YOLO11n-obb
# Predict:         yolo predict task=obb model=yolo11n-obb.torchscript imgsz=1024
# Validate:        yolo val task=obb model=yolo11n-obb.torchscript imgsz=1024 data=runs/DOTAv1-ms.yaml
# Visualize:       https://netron.app/

# 'yolo11n-obb.torchscript'

model = torch.jit.load("yolo11n-obb.torchscript")  # modèle torchscript, pas YOLO()
model.eval()

# Dummy input
x = torch.randn(1, 3, 1024, 1024)

with torch.no_grad():
    y = model(x)
    print(y.shape)
torch.Size([1, 20, 21504])
```

Could you clarify what the 20 channels represent in this case?
Is the angle or another component missing from the output?

Thanks in advance for your help!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @0xF9M, thank you for your detailed report and for providing a clear reproduction example! 🚀 This is an automated response to help get you the best support as quickly as possible—an Ultralytics engineer will also review and assist you soon.

It looks like you’re raising a question about the output shape of the YOLO11-OBB model and included a minimal reproducible example (MRE), which is super helpful—thank you! If there are any other details you think might be relevant (such as custom modifications or logs), please feel free to add them.

For reference, we recommend checking out the [Docs](https://docs.ultralytics.com/), where you’ll find [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, as well as information on [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/).

Join the Ultralytics community for real-time chat on [Discord](https://discord.com/invite/ultralytics) 🎧, deeper discussions on [Discourse](https://community.ultralytics.com/), or knowledge sharing on our [Subreddit](https://www.reddit.com/r/Ultralytics/).

## Upgrade

Please ensure you’re running the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. These tests verify that all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) run correctly across macOS, Windows, and Ubuntu on every commit and every 24 hours.

Thanks again for your detailed question and MRE! An Ultralytics engineer will review and provide further guidance soon.

**Comment by 0xF9M** [comment]
nvm my bad, I just forgot to set `nms=True` during export.

**Comment by Y-T-G** [comment]
There's no separate confidence score 

https://www.reddit.com/r/Ultralytics/comments/1jwl6se/ultralytics_postprocessing_guide/
